### PR TITLE
dbqn, dbqn-native: unstable-2021-10-08 -> 0.2.1

### PR DESCRIPTION
**Pull request by sternenseemann**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/interpreters/bqn/dzaima-bqn/default.nix [hunk 1]
```diff
@@ -8,13 +8,13 @@
...
-  version = "0.pre+date=2021-10-08";
+  version = "0.2.1";
 
   src = fetchFromGitHub {
     owner = "dzaima";
     repo = "BQN";
-    rev = "0001109a1c5a420421b368c79d34b1e93bfe606e";
-    hash = "sha256-riHHclTLkrVbtzmcz9ungAIc7kaoFHS77+SNatsfNhc=";
+    rev = "v0.2.1";
+    sha256 = "1kxzxz2hrd1871281s4rsi569qk314aqfmng9pkqn8gv9nqhmph0";
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/bqn/dzaima-bqn/default.nix:17]:
```suggestion
    hash = "sha256-AN4KsU37IYvnTc9WhxUJY+JkStSZ6IBEOCi0DMXvv88=
";
```

File: pkgs/development/interpreters/bqn/dzaima-bqn/default.nix [hunk 2]
```diff
@@ -8,13 +8,13 @@
...
   pname = "dbqn" + lib.optionalString buildNativeImage "-native";
-  version = "0.pre+date=2021-10-08";
+  version = "0.2.1";
 
   src = fetchFromGitHub {
     owner = "dzaima";
     repo = "BQN";
-    rev = "0001109a1c5a420421b368c79d34b1e93bfe606e";
-    hash = "sha256-riHHclTLkrVbtzmcz9ungAIc7kaoFHS77+SNatsfNhc=";
+    rev = "v0.2.1";
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/bqn/dzaima-bqn/default.nix:16]:
```suggestion
    rev = "v0.2.1";
```
```suggestion
    rev = "v${version}";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/interpreters/bqn/dzaima-bqn/default.nix (hunk 1 above)
Comment (AndersonTorres) [inline pkgs/development/interpreters/bqn/dzaima-bqn/default.nix:17]:
```suggestion
    rev = "v${version}";
    hash = "sha256-AN4KsU37IYvnTc9WhxUJY+JkStSZ6IBEOCi0DMXvv88=";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by AndersonTorres** [comment]
Mental note: Darwin is failing (again).

File: pkgs/development/interpreters/bqn/dzaima-bqn/default.nix (hunk 1 above)
Comment (sternenseemann) [reply pkgs/development/interpreters/bqn/dzaima-bqn/default.nix:17]: Done.

**Pull request merged**